src/pytorch/neighbors/getNeighborPairsCUDA.cu

```diff
@@ -100,6 +114,12 @@ public:
         TORCH_CHECK(max_num_neighbors_ > 0 || max_num_neighbors_ == -1,
             "Expected \"max_num_neighbors\" to be positive or equal to -1");
 
+        const bool use_periodic = (box_vectors.size(0) != 0);
+        if (use_periodic) {
```

Comment:
It could check here if all the requirements are satisfied too.

Reply:
Is there any way to check it efficiently?  The shape of the tensor is known by the CPU, but the values of its elements are stored on the GPU.  If we want to throw an exception based on the values of elements, that will require device to host data transfers and add significant latency every time it's called.

Reply:
Yes, you are right, it won't be efficient. So the checks have to be done on a GPU.

The only problem is how to abort a kernel elegantly. `assert ` does the jobs, but later a GPU needs a reset. In the CUDA docs, I don't see anything better.

Reply:
GPU asserts aren't really effective as a way of catching user errors.  In addition to the fact that you can't recover from them and have to reset the whole device, they don't provide any useful information to the user.  They just get a cryptic "device-side assert triggered" message that tells nothing about what the problem was or how to fix it.  The user will usually conclude that your library is broken.

Reply:
I agree, we need to choose between two evils:
- Users get cryptic messages and need to reset the GPU.
- Users get incorrect results silently.